src/main/java/com/softserveinc/dokazovi/service/impl/AuthorServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+package com.softserveinc.dokazovi.service.impl;
+
+import com.softserveinc.dokazovi.dto.author.AuthorForAdminDTO;
+import com.softserveinc.dokazovi.dto.author.AuthorDTO;
+import com.softserveinc.dokazovi.entity.CityEntity;
+import com.softserveinc.dokazovi.entity.DoctorEntity;
+import com.softserveinc.dokazovi.entity.RoleEntity;
+import com.softserveinc.dokazovi.entity.UserEntity;
+import com.softserveinc.dokazovi.entity.enumerations.RolePermission;
+import com.softserveinc.dokazovi.entity.enumerations.UserPromotionLevel;
+import com.softserveinc.dokazovi.entity.enumerations.UserStatus;
+import com.softserveinc.dokazovi.exception.ForbiddenPermissionsException;
+import com.softserveinc.dokazovi.repositories.CityRepository;
+import com.softserveinc.dokazovi.repositories.DoctorRepository;
+import com.softserveinc.dokazovi.repositories.UserRepository;
+import com.softserveinc.dokazovi.security.UserPrincipal;
+import com.softserveinc.dokazovi.service.AuthorService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Service
+@RequiredArgsConstructor
+public class AuthorServiceImpl implements AuthorService {
+
+	private final DoctorRepository doctorRepository;
+	private final CityRepository cityRepository;
+	private final PasswordEncoder passwordEncoder;
+	private final UserRepository userRepository;
+
+
+	@Override
+	public AuthorDTO save(AuthorDTO authorDTO, UserPrincipal userPrincipal) {
+		if (doctorRepository.findById(authorDTO.getId()).isPresent()) {
+			update(authorDTO, userPrincipal);
+		}
+		AuthorDTO dto;
+		if (userPrincipal.getAuthorities().stream()
+				.anyMatch(
+						grantedAuthority -> grantedAuthority.getAuthority()
+								.equals(RolePermission.CREATE_AUTHOR.getAuthority()))) {
```

Comment:
Please fix indentations. And let's extract variable `hasEnoughAuthorities` or `hasCreateAuthorAuthority` before if statement and put it in if-check instead of long expression.

`boolean hasCreateAuthorAuthority = userPrincipal.getAuthorities().stream()
				.anyMatch(grantedAuthority -> grantedAuthority.getAuthority()
								.equals(RolePermission.CREATE_AUTHOR.getAuthority()))`

Reply:
I'd suggest creating a universal private method `hasAuthority(UserPrincipal userPrincipal, RolePermission permision)`  that contains the following logic `userPrincipal.getAuthorities().stream()
				.anyMatch(
						grantedAuthority -> grantedAuthority.getAuthority()
								.equals(RolePermission.CREATE_AUTHOR.getAuthority()))`

Reply:
fixed